Library/Homebrew/language/node.rb

```diff
@@ -1,10 +1,44 @@
+require "json"
+
 module Language
   module Node
     def self.npm_cache_config
       "cache=#{HOMEBREW_CACHE}/npm_cache"
     end
 
-    def self.pack_for_installation
+    def self.node_system(cmd, *args)
+      ohai "#{cmd} #{args * " "}"
+      if ARGV.verbose?
+        safe_system(cmd, *args)
+      else
+        quiet_system(cmd, *args)
+        raise ErrorDuringExecution.new(cmd, args) unless $CHILD_STATUS.exitstatus.zero?
+      end
+    end
+
+    # Read https://gist.github.com/chrmoritz/34e4c4d7779d72b549e2fc41f77c365c
+    # for a complete overview of the edge cases this method has to handle.
+    def self.pack_for_installation(prepare_required: false)
+      # Rewrites the package.json so that npm pack will bundle all deps
+      # into a self-contained package, so that we avoid installing them a
+      # second time during the final installation of the package to libexec.
+      pkg_json = JSON.parse(IO.read("package.json"))
```

Comment:
I was thinking any error message that's not a raw exception but you've convinced me this is fine as-is.